### PR TITLE
Upgrade ava

### PR DESCRIPTION
**Pull request by SimenB**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,13 +52,16 @@
...
+    "statements": 98,
+    "functions": 97,
+    "branches": 93,
     "include": [
       "lib/**.js"
     ],
     "all": true
+  },
+  "ava": {
+    "failWithoutAssertions": false
```
Comment (sveisvei) [inline package.json]: I kinda like this being on though... you think its annoying?

Comment (SimenB) [inline package.json]: There is a test without assertions now

Comment (sveisvei) [inline package.json]: then use `t.plan(0)` :)

**Comment by SimenB** [comment]
One failing test, unsure what's up with it?

**Comment by sveisvei** [comment]
@SimenB force pushing here fyi.

**Comment by sveisvei** [comment]
@SimenB merge if you are OK with the `process.env.NODE_ENV=test` solution

**Comment by SimenB** [comment]
Hah, this was https://github.com/avajs/ava/issues/1484?

File: package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "lint:fix": "eslint . --fix  && prettier --write '{src,examples}/**/*.ts' --single-quote --tab-width 4 --print-width 100 --trailing-comma all --parser typescript",
     "build": "tsc src/*.ts src/**/*.ts -d --strictNullChecks --sourceMap --outdir ./lib --lib ES6,ES5",
     "prepublish": "not-in-install && npm run build || in-install",
-    "test": "ava test",
+    "test": "NODE_ENV=test ava test",
```
Comment (SimenB) [inline package.json]: Won't work on windows without doing `cross-env` or something. maybe care

Comment (sveisvei) [reply package.json]: I dont - but would if I knew of anyone running these tests on a windows env.

**Pull request merged**